include/fdreadoutlibs/wib2/tpg/ProcessRSAVX2.hpp

```diff
@@ -0,0 +1,394 @@
+/**
+ * @file ProcessRSAVX2.hpp Process frames with AVX2 registers and instructions
+ * using the Running Sum algorithm
+ * 
+ * @author Adam Abed Abud (adam.abed.abud@cern.ch)
+ *
+ * This is part of the DUNE DAQ , copyright 2022.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#ifndef READOUT_SRC_WIB2_TPG_PROCESSRSAVX2_HPP_
+#define READOUT_SRC_WIB2_TPG_PROCESSRSAVX2_HPP_
+
+#include "FrameExpand.hpp"
+#include "ProcessingInfo.hpp"
+#include "TPGConstants_wib2.hpp"
+
+#include <immintrin.h>
+
+namespace swtpg_wib2 {
+
+inline __m256i _mm256_div_epi16 (const __m256i va, const int b)
+{
+    __m256i vb = _mm256_set1_epi16(32768 / b);
+    return _mm256_mulhrs_epi16(va, vb);
+}
+
+
+
+inline void
+frugal_accum_update_rs_avx2(__m256i& __restrict__ median,
+                         const __m256i s,
+                         __m256i& __restrict__ accum,
+                         const int16_t acclimit,
+                         const __m256i mask) __attribute__((always_inline));
+
+inline void
+frugal_accum_update_rs_avx2(__m256i& __restrict__ median,
```

Comment:
Do we need to have 2 frugal accumulator functions? Wouldn't it be better to split it into a separate file and have the 2 process_window functions refer to the same one?

Reply:
Thanks, I'll make the fix shortly

Reply:
I pushed a fix for that